database/engine/rrdengineapi.c

```diff
@@ -603,21 +469,21 @@ static int rrdeng_load_page_next(struct rrddim_query_handle *rrdimm_handle) {
 // Returns the metric and sets its timestamp into current_time
 // IT IS REQUIRED TO **ALWAYS** SET ALL RETURN VALUES (current_time, end_time, flags)
 // IT IS REQUIRED TO **ALWAYS** KEEP TRACK OF TIME, EVEN OUTSIDE THE DATABASE BOUNDARIES
-NETDATA_DOUBLE
-rrdeng_load_metric_next(struct rrddim_query_handle *rrdimm_handle, time_t *start_time, time_t *end_time, SN_FLAGS *flags) {
+STORAGE_POINT rrdeng_load_metric_next(struct rrddim_query_handle *rrdimm_handle) {
```

Comment:
I'm curious if returning STORAGE_POINT will hurt performance significantly. I haven't checked x86_64's ABI, but my guess is that `STORAGE_POINT` values can't get materialized in registers.

Reply:
I guess this happens through the stack. Right?
I have not tested it and it is a really simple change to pass it as a pointer to the function, but then the function will have to dereference it for each of its members. Can you test the difference?